packages/inferno/src/DOM/utils/common.ts

```diff
@@ -80,12 +90,33 @@ export function findDOMfromVNode(vNode: VNode, startEdge: boolean) {
   return null;
 }
 
-export function removeVNodeDOM(vNode: VNode, parentDOM: Element) {
+export function callAllAnimationHooks(animationQueue: Function[], callback?: Function) {
+  let animsLeft = animationQueue.length;
+  do {
+    const fn = animationQueue.pop();
+    // This check shouldn't be needed but TS complains so adding it
+    if (fn !== undefined) {
```

Comment:
its needed because last pop will give empty, so the loop here is little bit wierd... you could do while(pop() ...)   or then just regular for loop if we need to calculate animations left

Reply:
is there any reason to actally "pop" from the array cant we just do linear - read ? simple for loop ?

Reply:
the issue with the latest loop is that its reading length of the array and the same array is mutated using pop so length does not get cached. 

We can work around by not reading length.
shift for first to last and pop for last to first
```js
 let listener;
  while ((callback = array.shift()) !== undefined) {
    callback();
  }
```
but regular for loop will be way faster if we do not need to "empty" the array

Reply:
or, if we know that there cannot be new items added to the array during the loop, but we need to empty it then we can do regular for loop and then at the end of the loop, clear it by reference  

array.length = 0;

Reply:
@Havunen the reason for .pop(): when multiple removes are triggered in the same patch you get multiple calls to callAllAnimationHooks. I tried a for loop but it failed during tests and no elements where removed. I couldn't really figure out what was happening because I can't step through the code properly (misaligned cursor during debugging). 

Implemented you suggested optimisation of the while loop.

FIXED in ca9b1277a